miden/src/cli/run.rs

```diff
@@ -1,20 +1,30 @@
-use std::{path::PathBuf, time::Instant};
+use std::{fs, path::PathBuf, time::Instant};
 
 use assembly::diagnostics::{IntoDiagnostic, Report, WrapErr};
-use clap::Parser;
+use clap::{ArgGroup, Parser};
 use miden_vm::internal::InputFile;
+use package::Package;
 use processor::{DefaultHost, ExecutionOptions, ExecutionTrace};
+use prover::utils::Deserializable;
 use stdlib::StdLibrary;
 use tracing::instrument;
 
 use super::data::{Libraries, OutputFile, ProgramFile};
 
 #[derive(Debug, Clone, Parser)]
-#[clap(about = "Run a miden program")]
+#[clap(about = "Run a miden program", group(
+    ArgGroup::new("input_source")
+        .required(true)
+        .args(&["assembly_file", "package_file"])
+))]
 pub struct RunCmd {
     /// Path to .masm assembly file
     #[clap(short = 'a', long = "assembly", value_parser)]
-    assembly_file: PathBuf,
+    assembly_file: Option<PathBuf>,
+
+    /// Path to .masp package file
+    #[clap(short = 'p', long = "package", value_parser)]
+    package_file: Option<PathBuf>,
```

Comment:
I think it would be more natural to make the file to execute be a positional parameter rather than requiring a flag. So you would type e.g. `miden run my_file.masm` or `miden run my_pkg.masp`, and we would figure out based on extension (or optionally also the ability to parse if no extension is provided) which of the 2 cases it is.

Same comment applies to the `prove` command. There we also wouldn't need to rename the other parameters.

Reply:
Yes, I think think updating this to use positional parameters would work very well here.

Reply:
And for now, to keep things simple, I'd figure out which file it is by just looking at the extension.